tikv-client/src/main/java/com/pingcap/tikv/region/TiRegion.java

```diff
@@ -124,8 +141,8 @@ boolean switchPeer(long leaderStoreID) {
   }
 
   public boolean contains(ByteString key) {
-    return meta.getStartKey().equals(key)
-        && (meta.getEndKey().equals(key) || meta.getEndKey().isEmpty());
+    return (meta.getStartKey().toStringUtf8().compareTo(key.toStringUtf8()) <=0 || meta.getStartKey().isEmpty())
```

Comment:
This looks wrong to me. Once bytes are decoded as UTF8, essentially the content changed and hence sort order changed as well. Use FastByteComparisons instead.